src/main/kotlin/org/malachite/estella/services/SecurityService.kt

```diff
@@ -64,24 +62,23 @@ class SecurityService(
             .compact()
     }
 
-    private fun isSigned(jwt: String, secret: String) =
-        Jwts.parser()
-            .setSigningKey(secret)
-            .isSigned(jwt)
+    private fun isSigned(jwt: String, secret: String): Boolean =
+        Jwts.parser().setSigningKey(secret).isSigned(jwt)
 
     private fun parseJWT(jwt: String, secret: String) =
-        Jwts.parser()
-            .setSigningKey(secret)
-            .parseClaimsJws(jwt)
-
-
-    fun getUserFromJWT(jwt: String?, secret: String = authSecret): User? {
-        if (jwt == null || !isSigned(jwt, secret)) return null
-        val id = parseJWT(jwt, secret)
-            .body
-            .issuer
-        return userRepository.findById(id.toInt()).orElse(null)
-    }
+        Jwts.parser().setSigningKey(secret).parseClaimsJws(jwt)
+
+    fun getUserFromJWT(jwt: String?, secret: String = authSecret): User? =
+        try {
+            if (jwt == null || !isSigned(jwt, secret))
+                null
+            else
+                parseJWT(jwt, secret).body
+                    .issuer
+                    .let { userRepository.findById(it.toInt()).orElse(null)}
+        } catch (ex: SignatureException) {
```

Comment:
pierdoła, ale mógłbyś wypisać stack trace'a - jak walnie, będziemy wiedzieli co

Reply:
Hmm ale to nam strasznie zawali logi w recruitment Service bo będzie ci wypisywało stackTrace za każdym razem jak będziesz przesyłał endpoint z obcego punktu i takie np puszczenie testów z MeetingOrganizera, to będzie 30 razy stack trace XDDD